test/FsAutoComplete.Tests.Lsp/CodeFixTests/Tests.fs

```diff
@@ -1637,6 +1637,180 @@ let private generateXmlDocumentationTests state =
             let x = 3
         """ ])
 
+let private addMissingXmlDocumentationTests state =
+  serverTestList (nameof AddMissingXmlDocumentation) state defaultConfigDto None (fun server ->
+    [ let selectCodeFix = CodeFix.withTitle AddMissingXmlDocumentation.title
+
+      testCaseAsync "missing params and returns for function with two parameters"
+      <| CodeFix.check
+        server
+        """
+        /// <summary>some comment$0</summary>
+        let f x y = x + y
+        """
+        Diagnostics.acceptAll
+        selectCodeFix
+        """
+        /// <summary>some comment</summary>
+        /// <param name="x"></param>
+        /// <param name="y"></param>
+        /// <returns></returns>
+        let f x y = x + y
+        """
+
+      testCaseAsync "missing params and returns for nested function with two parameters"
+      <| CodeFix.check
+        server
+        """
+        let f x y =
+          /// <summary>some comment$0</summary>
+          let g a b =
+            a + b
+          x + y
+        """
+        Diagnostics.acceptAll
+        selectCodeFix
+        """
+        let f x y =
+          /// <summary>some comment</summary>
+          /// <param name=""></param>
+          /// <param name=""></param>
+          /// <returns></returns>
+          let g a b =
+            a + b
+          x + y
+        """
+
+      testCaseAsync "missing single parameter for function with two parameters"
+      <| CodeFix.check
+        server
+        """
+        /// <summary>some comment$0</summary>
+        /// <param name="x"></param>
+        /// <returns></returns>
+        let f x y = x + y
+        """
+        Diagnostics.acceptAll
+        selectCodeFix
+        """
+        /// <summary>some comment</summary>
+        /// <param name="x"></param>
+        /// <param name="y"></param>
+        /// <returns></returns>
+        let f x y = x + y
+        """
+
+      testCaseAsync "missing type parameter for function"
+      <| CodeFix.check
+        server
+        """
+        /// <summary>some comment$0</summary>
+        /// <param name="x"></param>
+        /// <param name="y"></param>
+        /// <returns></returns>
+        let f x y _ = x + y
+        """
+        Diagnostics.acceptAll
+        selectCodeFix
+        """
+        /// <summary>some comment</summary>
+        /// <param name="x"></param>
+        /// <param name="y"></param>
+        /// <param name=""></param>
+        /// <typeparam name="'a"></typeparam>
+        /// <returns></returns>
+        let f x y _ = x + y
+        """
+
+      testCaseAsync "not applicable for function without summary"
+      <| CodeFix.checkNotApplicable
+        server
+        """
+        /// some comment$0
+        let f x y = x + y
+        """
+        Diagnostics.acceptAll
+        selectCodeFix
```

Comment:
How hard would it be to make this case work? It'd be really nice to have one gesture to go from 'simple' xmldoc to 'full' xmldoc.

Reply:
Oh, I'm afraid there was some misunderstanding on my side. I thought your goal was to have a two step approach like:
1. convert tripple slash comment to xml comment
2. add missing xml tags

I'll take a look. Shouldn't be too much work.

Reply:
It's in.


https://user-images.githubusercontent.com/3221269/234708698-e48a5cdd-c331-4896-9bdc-f11a0726e56c.mp4